### PR TITLE
Use more pcl::Indices in features, segmentation, tracking, ...

### PR DESCRIPTION
**Pull request by mvieth**

### CODE REVIEW

File: features/include/pcl/features/impl/ppfrgb.hpp [hunk 1]
```diff
@@ -125,7 +125,7 @@ pcl::PPFRGBRegionEstimation<PointInT, PointNT, PointOutT>::computeFeature (Point
   output.resize (indices_->size ());
   for (std::size_t index_i = 0; index_i < indices_->size (); ++index_i)
```
Comment (larshg) [inline features/include/pcl/features/impl/ppfrgb.hpp:126]: Could be ranged based for loop?

Comment (mvieth) [reply features/include/pcl/features/impl/ppfrgb.hpp:126]: Sadly no (at least not trivially): `index_i` is used in line 171

**Review by larshg — APPROVED** [review]

File: people/include/pcl/people/impl/head_based_subcluster.hpp [hunk 2]
```diff
@@ -214,9 +213,9 @@ pcl::people::HeadBasedSubclustering<PointT>::createSubClusters (pcl::people::Per
   }
 
   // Associate cluster points to one of the maximum:
-  for(std::vector<int>::const_iterator points_iterator = cluster.getIndices().indices.begin(); points_iterator != cluster.getIndices().indices.end(); ++points_iterator)
+  for(const auto& points_iterator : cluster.getIndices().indices)
```
Comment (kunaltyagi) [inline people/include/pcl/people/impl/head_based_subcluster.hpp:216]: Not an iterator

File: people/include/pcl/people/impl/head_based_subcluster.hpp [hunk 3]
```diff
@@ -214,9 +213,9 @@ pcl::people::HeadBasedSubclustering<PointT>::createSubClusters (pcl::people::Per
...
   {
-    PointT* current_point = &(*cloud_)[*points_iterator];        // current point cloud point
+    PointT* current_point = &(*cloud_)[points_iterator];        // current point cloud point
```
Comment (kunaltyagi) [inline people/include/pcl/people/impl/head_based_subcluster.hpp:218]: Pointer not needed

File: people/include/pcl/people/impl/head_based_subcluster.hpp [hunk 4]
```diff
@@ -177,10 +177,9 @@ pcl::people::HeadBasedSubclustering<PointT>::mergeClustersCloseInFloorCoordinate
           if (!used_clusters[cluster])         // if this cluster has not been used yet
           {
             used_clusters[cluster] = true;
-            for(std::vector<int>::const_iterator points_iterator = input_clusters[cluster].getIndices().indices.begin();
-                points_iterator != input_clusters[cluster].getIndices().indices.end(); ++points_iterator)
+            for(const auto& points_iterator : input_clusters[cluster].getIndices().indices)
```
Comment (kunaltyagi) [inline people/include/pcl/people/impl/head_based_subcluster.hpp:180]: Not an iterator

File: people/include/pcl/people/impl/height_map_2d.hpp [hunk 5]
```diff
@@ -85,9 +85,9 @@ pcl::people::HeightMap2D<PointT>::compute (pcl::people::PersonCluster<PointT>& c
     buckets_.resize(std::size_t((cluster.getMax()(1) - cluster.getMin()(1)) / bin_size_) + 1, 0);
   buckets_cloud_indices_.resize(buckets_.size(), 0);
 
-  for(std::vector<int>::const_iterator pit = cluster.getIndices().indices.begin(); pit != cluster.getIndices().indices.end(); ++pit)
+  for(const auto& pit : cluster.getIndices().indices)
```
Comment (kunaltyagi) [inline people/include/pcl/people/impl/height_map_2d.hpp:88]: Not an iterator

File: people/include/pcl/people/impl/head_based_subcluster.hpp (hunk 3 above)
Comment (mvieth) [reply people/include/pcl/people/impl/head_based_subcluster.hpp]: Removed.

File: people/include/pcl/people/impl/head_based_subcluster.hpp (hunk 2 above)
Comment (mvieth) [reply people/include/pcl/people/impl/head_based_subcluster.hpp]: Renamed (others as well)

**Review by kunaltyagi — APPROVED** [review]

**Comment by kunaltyagi** [comment]
🚀

**Pull request merged**